src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Definition/RazorDefinitionEndpoint.cs

```diff
@@ -0,0 +1,144 @@
+            }, cancellationToken, TaskCreationOptions.None, _foregroundDispatcher.ForegroundScheduler).ConfigureAwait(false);
+
+            if (documentSnapshot is null)
+            {
+                return null;
+            }
+
+            if (!FileKinds.IsComponent(documentSnapshot.FileKind))
+            {
+                return null;
+            }
+
+            var codeDocument = await documentSnapshot.GetGeneratedOutputAsync().ConfigureAwait(false);
+            if (codeDocument.IsUnsupported())
+            {
+                return null;
+            }
+
+            var originTagHelperBinding = await GetOriginTagHelperBindingAsync(documentSnapshot, codeDocument, request.Position).ConfigureAwait(false);
+            if (originTagHelperBinding is null)
+            {
+                return null;
+            }
+
+            var originTagDescriptor = originTagHelperBinding.Descriptors.FirstOrDefault();
+            if (originTagDescriptor is null)
+            {
+                return null;
+            }
+
+            var originComponentDocumentSnapshot = await _componentSearchEngine.TryLocateComponentAsync(originTagDescriptor).ConfigureAwait(false);
+            if (originComponentDocumentSnapshot is null)
+            {
+                return null;
+            }
+
+            var originComponentUri = new UriBuilder
+            {
+                Path = originComponentDocumentSnapshot.FilePath,
+                Scheme = Uri.UriSchemeFile,
+                Host = string.Empty,
+            }.Uri;
+
+            return new LocationOrLocationLinks(new[]
+            {
+                new LocationOrLocationLink(new Location
+                {
+                    Uri = originComponentUri,
+                    Range = new Range(new Position(0, 0), new Position(0, 0)),
+                }),
+            });
+        }
+
+        public void SetCapability(DefinitionCapability capability)
+        {
+            _capability = capability;
+        }
+
+        private async Task<TagHelperBinding> GetOriginTagHelperBindingAsync(DocumentSnapshot documentSnapshot, RazorCodeDocument codeDocument, Position position)
+        {
+            var sourceText = await documentSnapshot.GetTextAsync().ConfigureAwait(false);
+            var linePosition = new LinePosition((int)position.Line, (int)position.Character);
+            var hostDocumentIndex = sourceText.Lines.GetPosition(linePosition);
+            var location = new SourceLocation(hostDocumentIndex, (int)position.Line, (int)position.Character);
+
+            var change = new SourceChange(location.AbsoluteIndex, length: 0, newText: string.Empty);
+            var syntaxTree = codeDocument.GetSyntaxTree();
+            if (syntaxTree?.Root is null)
+            {
+                return null;
+            }
+
+            var owner = syntaxTree.Root.LocateOwner(change);
+            var node = owner.Ancestors().FirstOrDefault(n => n.Kind == SyntaxKind.MarkupTagHelperStartTag);
```

Comment:
This means if you F12 anywhere inside the start tag it'll bring you to the component right? Why not see if the location intersects with the start tag's name?

For instance imagine you had: `<Counter IncrementAmount="MyConstants.DefaultIncrement" />` and you F12'd in `DefaultIncrement`. With how you have it right now I believe you'd get 2 go-to-definition results. One from C# for DefaultIncrement and the other to Counter.razor. We'd expect to only get 1 in that situation.

Reply:
Whoops, good call.